lib/compiler/passes/generate-bytecode.js

```diff
@@ -535,6 +554,8 @@ function generateBytecode(ast) {
         [op.PUSH_EMPTY_ARRAY],
         expressionCode,
         buildCondition(
+          // Check expression match, not the node match
+          node.expression.match | 0,
```

Comment:
why is this random code repeated all over the place?

why isn't it behind a method with a clear name that explains what it does?

Reply:
Explained above.

Reply:
No, it wasn't.  You gave a Zorba the Mystic "when you look deep enough into the soul of `match` you will understand."